#VRML_SIM R2020b utf8
WorldInfo {
  basicTimeStep 2
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.469177790538435 0.8726210195835791 0.13566413323493978 0.5914986329094769
  position 10.140686502008831 5.979675325062639 15.792727810255263
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 60 90
  floorTileSize 1.5 1.5
  wallHeight 0.001
}
Robot {
  translation 0 0.8 0
  children [
    Hinge2Joint {
      jointParameters HingeJointParameters {
      }
      jointParameters2 JointParameters {
      }
      device [
        RotationalMotor {
          name "X rotational motor B"
          minPosition -0.5
          maxPosition 0.5
          maxTorque 1000
        }
        PositionSensor {
          name "X position sensor B"
        }
      ]
      device2 [
        RotationalMotor {
          name "Z rotational motor B"
          maxTorque 1000
        }
        PositionSensor {
          name "Z position sensor B"
        }
      ]
      endPoint Solid {
        translation -3.132605147929815e-13 0.00022956194274170732 -1.2420957273116862e-20
        children [
          SliderJoint {
            jointParameters JointParameters {
              axis 0 1 0
            }
            device [
              PositionSensor {
                name "Shorten position sensor B"
              }
              LinearMotor {
                name "Shorten linear motor B"
                maxForce 1000
              }
            ]
            endPoint Solid {
              children [
                Transform {
                  rotation 1 0 0 -1.5708
                  children [
                    Shape {
                      appearance PBRAppearance {
                        baseColor 0 0 1
                      }
                      geometry Cylinder {
                        height 1
                        radius 0.01
                      }
                      isPickable FALSE
                    }
                  ]
                }
                SliderJoint {
                  jointParameters JointParameters {
                    axis 0 1 0
                    minStop -0.4
                    dampingConstant 10
                    staticFriction 0.2
                  }
                  device [
                    LinearMotor {
                      name "Spring linear motor B"
                      maxForce 1600
                    }
                    PositionSensor {
                      name "Spring position sensor B"
                    }
                  ]
                  endPoint Solid {
                    children [
                      Transform {
                        rotation 1 0 0 -1.5708
                        children [
                          Shape {
                            appearance PBRAppearance {
                              baseColor 0 0 1
                            }
                            geometry Cylinder {
                              height 1
                              radius 0.01
                            }
                            isPickable FALSE
                          }
                        ]
                      }
                      DEF leg Shape {
                        appearance PBRAppearance {
                          baseColor 1 1 0
                          metalness 0
                        }
                        geometry Cylinder {
                          height 1.58
                          radius 0.02
                        }
                        castShadows FALSE
                      }
                      Transform {
                        translation 0 -0.77 0
                        children [
                          DEF foot Shape {
                            appearance PBRAppearance {
                              baseColor 1 0 0
                              metalness 0
                            }
                            geometry Sphere {
                              radius 0.03
                              subdivision 3
                            }
                            castShadows FALSE
                          }
                          TouchSensor {
                            name "touch sensor B"
                            boundingObject USE foot
                            physics Physics {
                            }
                          }
                        ]
                      }
                    ]
                    boundingObject USE leg
                    physics Physics {
                      density -1
                      mass 0.5
                    }
                  }
                }
              ]
              name "move b solid"
              boundingObject Sphere {
                radius 0.01
              }
              physics Physics {
                density -1
                mass 0.001
              }
            }
          }
        ]
        name "solid B"
        boundingObject DEF shaft Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Cylinder {
            height 0.3
            radius 0.05
          }
          castShadows FALSE
        }
        physics Physics {
          density -1
          mass 0.25
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "textures/water.jpg"
          ]
        }
        metalness 0
      }
      geometry Cylinder {
        height 0.1
        radius 0.6
      }
      castShadows FALSE
    }
    Hinge2Joint {
      jointParameters HingeJointParameters {
      }
      jointParameters2 JointParameters {
      }
      device [
        RotationalMotor {
          name "X rotational motor A"
          minPosition -0.5
          maxPosition 0.5
          maxTorque 1000
        }
        PositionSensor {
          name "X position sensor A"
        }
      ]
      device2 [
        RotationalMotor {
          name "Z rotational motor A"
          maxTorque 1000
        }
        PositionSensor {
          name "Z position sensor A"
        }
      ]
      endPoint Solid {
        translation -3.132605147929815e-13 0.00022956194274170732 -1.2420957273116862e-20
        children [
          Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                }
                geometry Cylinder {
                  radius 0.01
                }
                isPickable FALSE
              }
            ]
          }
          Transform {
            rotation 1 0 0 -1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                }
                geometry Cylinder {
                  radius 0.01
                }
                isPickable FALSE
              }
            ]
          }
          SliderJoint {
            jointParameters JointParameters {
              axis 0 1 0
            }
            device [
              PositionSensor {
                name "Shorten position sensor A"
              }
              LinearMotor {
                name "Shorten linear motor A"
                maxForce 1000
              }
            ]
            endPoint Solid {
              children [
                Transform {
                  rotation 1 0 0 -1.5708
                  children [
                    Shape {
                      appearance PBRAppearance {
                        baseColor 1 0 0
                      }
                      geometry Cylinder {
                        height 1
                        radius 0.01
                      }
                      isPickable FALSE
                    }
                  ]
                }
                SliderJoint {
                  jointParameters JointParameters {
                    axis 0 1 0
                    minStop -0.4
                    dampingConstant 10
                    staticFriction 0.2
                  }
                  device [
                    LinearMotor {
                      name "Spring linear motor A"
                      maxForce 1600
                    }
                    PositionSensor {
                      name "Spring position sensor A"
                    }
                  ]
                  endPoint Solid {
                    children [
                      Transform {
                        rotation 0 0 1 1.5708
                        children [
                          Shape {
                            appearance PBRAppearance {
                              baseColor 1 0 0
                            }
                            geometry Cylinder {
                              height 1
                              radius 0.01
                            }
                            isPickable FALSE
                          }
                        ]
                      }
                      Transform {
                        rotation 1 0 0 -1.5708
                        children [
                          Shape {
                            appearance PBRAppearance {
                              baseColor 0 0 1
                            }
                            geometry Cylinder {
                              height 1
                              radius 0.01
                            }
                            isPickable FALSE
                          }
                        ]
                      }
                      DEF leg Shape {
                        appearance PBRAppearance {
                          baseColor 1 1 0
                          metalness 0
                        }
                        geometry Cylinder {
                          height 1.58
                          radius 0.02
                        }
                        castShadows FALSE
                      }
                      Transform {
                        translation 0 -0.77 0
                        children [
                          DEF foot Shape {
                            appearance PBRAppearance {
                              baseColor 1 0 0
                              metalness 0
                            }
                            geometry Sphere {
                              radius 0.03
                              subdivision 3
                            }
                            castShadows FALSE
                          }
                          TouchSensor {
                            name "touch sensor A"
                            boundingObject USE foot
                            physics Physics {
                            }
                          }
                        ]
                      }
                    ]
                    boundingObject USE leg
                    physics Physics {
                      density -1
                      mass 0.5
                    }
                  }
                }
              ]
              name "move a solid"
              boundingObject Sphere {
                radius 0.01
              }
              physics Physics {
                density -1
                mass 0.001
              }
            }
          }
        ]
        boundingObject DEF shaft Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Cylinder {
            height 0.3
            radius 0.05
          }
          castShadows FALSE
        }
        physics Physics {
          density -1
          mass 0.25
        }
      }
    }
    InertialUnit {
      boundingObject USE body
      physics Physics {
        density -1
        mass 0.5
      }
    }
  ]
  boundingObject USE body
  physics Physics {
    density -1
    mass 14
  }
  controller "my_controller_2_leg"
}
Robot {
  translation 4.95 2.49 -0.86
  children [
    DEF externBody Shape {
      geometry Sphere {
        radius 0.1
      }
      castShadows FALSE
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 0.2
      }
      endPoint Solid {
        translation 0 -1.4701000000000148 -0.29800400000000327
        rotation 1 0 0 0.1999999999999989
        children [
          DEF externBall Shape {
            appearance PBRAppearance {
              roughness 0.5
            }
            geometry Sphere {
              radius 0.2
              subdivision 3
            }
            castShadows FALSE
          }
          Transform {
            translation 0 0.75 0
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 0.5
                  metalness 0
                  IBLStrength 2
                }
                geometry Cylinder {
                  height 1.5
                  radius 0.01
                }
                castShadows FALSE
              }
            ]
          }
        ]
        boundingObject USE externBall
        physics Physics {
          density -1
          mass 2
        }
      }
    }
  ]
  name "robot(1)"
  boundingObject USE externBody
}
